src/platform/packages/shared/as-code/filters-transforms/src/from_stored_filter.ts

```diff
@@ -0,0 +1,513 @@
+  } catch (error) {
+    logger?.warn(`Failed to convert stored filter: ${error.message}`);
+    return;
+  }
+}
+
+/**
+ * Convert stored filter to simple condition
+ * Extracts field, operator, and value from filter metadata and query structures
+ * Used by type-specific converters and as a helper for group conversion
+ */
+function convertToSimpleCondition(storedFilter: StoredFilter): AsCodeConditionFilter['condition'] {
+  const meta = storedFilter.meta || {};
+  const query = storedFilter.query || {};
+
+  // Extract field name - ExtendedFilter type includes optional 'field' property
+  const field = meta.key || meta.field || extractFieldFromQuery(query);
+  if (!field) {
+    throw new FilterConversionError('Cannot determine field name from stored filter');
+  }
+
+  // Determine operator and value based on query structure
+  if (hasExistsQuery(query)) {
+    return {
+      field,
+      operator: meta.negate ? ASCODE_FILTER_OPERATOR.NOT_EXISTS : ASCODE_FILTER_OPERATOR.EXISTS,
+    };
+  }
+
+  if (hasRangeQuery(query)) {
+    const range = query.range[field] as Record<string, unknown>;
+    return {
+      field,
+      operator: ASCODE_FILTER_OPERATOR.RANGE,
+      value: {
+        ...(range.gte !== undefined && { gte: range.gte as number | string }),
+        ...(range.lte !== undefined && { lte: range.lte as number | string }),
+        ...(range.gt !== undefined && { gt: range.gt as number | string }),
+        ...(range.lt !== undefined && { lt: range.lt as number | string }),
+        ...(range.format !== undefined && { format: range.format as string }),
+      },
+    };
+  }
+
+  // Handle phrases filter: meta.type='phrases' with bool.should containing match_phrase clauses
+  if (isPhrasesFilter(storedFilter)) {
+    // Extract values from match_phrase clauses
+    const values = query.bool.should
+      .map((clause: unknown) => {
+        if (typeof clause === 'object' && clause !== null && 'match_phrase' in clause) {
+          const matchPhrase = (clause as Record<string, unknown>).match_phrase as Record<
+            string,
+            unknown
+          >;
+          const clauseField = Object.keys(matchPhrase)[0];
+          return matchPhrase[clauseField];
+        }
+        return;
+      })
+      .filter((v: unknown) => v !== undefined) as Array<string | number | boolean>;
+
+    // Validate homogeneous array
+    validateHomogeneousArray(values, 'Phrases filter');
+
+    return {
+      field,
+      operator: meta.negate
+        ? ASCODE_FILTER_OPERATOR.IS_NOT_ONE_OF
+        : ASCODE_FILTER_OPERATOR.IS_ONE_OF,
+      value: values as string[] | number[] | boolean[],
+    };
+  }
+
+  if (query.term) {
+    const value = query.term[field];
+    return {
+      field,
+      operator: meta.negate ? ASCODE_FILTER_OPERATOR.IS_NOT : ASCODE_FILTER_OPERATOR.IS,
+      value,
+    };
+  }
+
+  if (query.match) {
+    const matchField = Object.keys(query.match)[0];
```

Comment:
Same question here, I was under the impression we always generate phrase queries.